spring-grpc-core/src/main/java/org/springframework/grpc/server/DefaultGrpcServerFactory.java

```diff
@@ -113,38 +120,35 @@ else if (ShadedNettyServerFactoryHelper.isEpollAvailable()) {
 	}
 
 	/**
-	 * Configures the given server builder. This method can be overwritten to add features
-	 * that are not yet supported by this library or use a {@link GrpcServerConfigurer}
-	 * instead.
-	 * @param builder The server builder to configure.
+	 * Configures the server builder by adding service definitions and applying
+	 * customizers to the builder.
+	 * <p>
+	 * Subclasses can override this to add features that are not yet supported by this
+	 * library.
+	 * @param builder the server builder to configure
+	 * @param serviceDefinitions the service definitions to add to the builder
 	 */
-	protected void configure(final T builder) {
-		configureServices(builder);
-		for (final GrpcServerConfigurer serverConfigurer : this.serverConfigurers) {
-			serverConfigurer.accept(builder);
-		}
+	protected void configure(T builder, List<ServerServiceDefinition> serviceDefinitions) {
```

Comment:
Added the service defs to the contract as subclasses could not override and access the service defs.